### PR TITLE
executors: add compiler syscalls option to CompiledExecutor

### PR DESCRIPTION
**Pull request by Riolku**

### CODE REVIEW

File: dmoj/executors/compiled_executor.py [hunk 1]
```diff
@@ -104,6 +106,20 @@ def get_compile_env(self) -> Optional[Dict[str, str]]:
     def get_compile_popen_kwargs(self) -> Dict[str, Any]:
         return {}
 
+    def _add_compiler_syscalls(self, sec: CompilerIsolateTracer) -> CompilerIsolateTracer:
```
Comment (Xyene) [inline dmoj/executors/compiled_executor.py:109]: This is a copy of `_add_syscalls`. Let's just make that take the list of syscalls to add, and use it here.

**Review by Xyene — CHANGES_REQUESTED** [review]

File: dmoj/executors/compiled_executor.py (hunk 1 above)
Comment (Riolku) [reply dmoj/executors/compiled_executor.py]: I completely misunderstood, let me try again.

Comment (Riolku) [reply dmoj/executors/compiled_executor.py]: Alright, should be good now.

**Review by Xyene — APPROVED** [review]
LGTM
